.github/workflows/dev.yml

```diff
@@ -1,57 +1,61 @@
 name: Development
 on: [pull_request, workflow_dispatch]
 
+env: 
+  # Override the `DOCKER_IMAGE` variable in Makefile.docker  
+  # This makes all Makefile targets use the specified Docker image instead of building a new one.
+  DOCKER_IMAGE: datadog/docker-library:dd-policy-engine-a0f48dd
+  BUILD_CONTAINER: false
+
 jobs:
   checks:
     runs-on: ubuntu-22.04
     timeout-minutes: 5
-    container:
-      image: datadog/docker-library:dd-policy-engine-a0f48dd
-    env:
-      IN_DOCKER: "true"
     permissions:
       contents: read
       packages: read
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - run: make fmt-check
 
-  # TODO: Add GCC compiler
   build-linux:
     strategy:
       fail-fast: false
       matrix:
         toolchain: [gcc, clang]
+        arch: [x86, arm64]
+        include:
+          - arch: x86
+            runner: ubuntu-22.04
+          - arch: arm64
+            runner: ubuntu-22.04-arm
     needs: checks
-    runs-on: ubuntu-22.04
+    runs-on: ${{ matrix.runner }}
     timeout-minutes: 10
-    container:
-      image: datadog/docker-library:dd-policy-engine-a0f48dd
     environment:
       name: dev-env
     permissions:
       contents: read
       packages: read
     env:
-      IN_DOCKER: "true"
       DD_ENV: ci
       DD_SERVICE: dd-policy-engine
       DD_CIVISIBILITY_AGENTLESS_ENABLED: true
     steps:
       - name: Checkout
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
       - name: Build
-        run: make test-build
-        working-directory: c
+        run: make docker-run "make -C c test-build"
         env:
           CMAKE_PRESET: ${{ matrix.toolchain }}-ci
-      - run: ctest --test-dir build --schedule-random --output-junit test-report.xml
-        working-directory: c
-      # - name: Report test coverage
-      #   uses: datadog/junit-upload-github-action@762867566348d59ac9bcf479ebb4ec040db8940a # v2.0.0
-      #   with:
-      #     api_key: ${{ secrets.DD_API_KEY_CI_VIS_APP }}
-      #     files: c/build/test-report.xml
+      - name: Run Tests
+        run: make docker-run "ctest --test-dir c/build --schedule-random --output-junit test-report.xml"
+      - name: Report test coverage
+        uses: datadog/junit-upload-github-action@762867566348d59ac9bcf479ebb4ec040db8940a # v2.0.0
+        with:
+          api_key: ${{ secrets.DD_API_KEY_CI_VIS_APP }}
+          files: c/build/test-report.xml
+          # tags: test.source.file:test/*.cpp
```

Comment:
why this comment? is it something we'll add later?

Reply:
yes!